rust/lance/src/dataset/write/insert.rs

```diff
@@ -265,7 +265,7 @@ impl<'a> InsertBuilder<'a> {
             WriteMode::Append => Operation::Append { fragments: blob.0 },
         });
 
-        Ok(Transaction::new(
+        let mut transaction = Transaction::new(
```

Comment:
since we added properties to the transaction proto, I think it makes more sense to just put properties as a part of the constructor.

Reply:
In that case, seems like we should develop a `TransactionBuilder`? But overall I think it's better to either use a builder or update the `new`, instead of making the struct mutable for the sake of setting the optional field.